lib/events.js

```diff
@@ -42,27 +42,74 @@ exports.onData = function onData(message) {
         // and be reissued again.
       }
     );
-  } else {
-    message.del();
   }
-};
 
-exports.onError = function onError(err) {
-  logger.error('accountEvent', err);
-};
+  function primaryEmailChanged(message) {
+    var userId = message.uid.split('@')[0];
+    if (! HEX_STRING.test(userId)) {
+      message.del();
+      return logger.warn('badPrimaryEmailChanged', {userId: userId});
+    }
 
-exports.start = function start() {
-  if (! config.events.region || ! config.events.queueUrl) {
-    if (env.isProdLike()) {
-      throw new Error('config.events must be included in prod');
+    // We currently a generate a user's cache key from their
+    // credentials in a request object. This builds an object
+    // to match what the server is expecting.
+    var req = {auth: {credentials: {user: userId}}};
+    P.resolve(new Promise(function(resolve) {
+      server.methods.batch.cache.drop(req, function () {
+        logger.info('primaryEmailChanged:cacheCleared', {uid: userId});
+        notifyProfileUpdated(userId);
+        resolve();
+      });
+    })).done(
```

Comment:
I wonder if factoring out this `done(...message.del()...)` logic into some common routine would make the code clearer and more robust, e.g. by having `deleteUser` and `primaryEmailChanged` return a promise, and having `onData` add a common cleanup block onto it.